bigframes/core/__init__.py

```diff
@@ -183,7 +183,7 @@ def project_to_id(self, expression: ex.Expression, output_id: str):
                 child=self.node,
                 assignments=tuple(exprs),
             )
-        )
+        ).rewrite_projection()
```

Comment:
how to pick up these functions and ask for rewriting the projection? Can we add the `rewrite_projection` to the ArrayValue constructor? In case other developers who will add new function here and miss the `rewrite_projection`  while missing context.

Reply:
moved to constructor in new revision.

Reply:
on second thought, this isn't really possible, as ArrayValue is an immutable dataclass, and mutation (by rewriting) is blocked even in the initializer

Reply:
I see. Thanks for trying that out! Maybe adding some comments for why we can add the `rewrite_projection` here? So other developers can have more context when they add something similar?